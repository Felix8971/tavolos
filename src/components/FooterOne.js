import React from 'react';

class FooterOne extends React.Component {
  constructor(props) {
    super();
  }
  
  render() {
    return (
      <div className='footer-one'>
        <div className='item'>
          <h4>Find out about new openings and offers.</h4>
          <p>Shh...Just don't tell anyone.</p>
        </div>
        <div className='item'>
          <form>
            <label>
              <input type="text" name="name" placeholder="email address"/>
            </label>
            <input type="submit" value="Go" />
          </form>
        </div>
      </div>
    )
  }
}

export default FooterOne;


