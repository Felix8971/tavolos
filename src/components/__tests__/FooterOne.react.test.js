import React from 'react';
import { shallow, mount, render } from 'enzyme';
import FooterOne from '../FooterOne';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

describe('FooterOne', () => {

    it('should be defined', () => {
      expect(FooterOne).toBeDefined();
    });

    it('should render correctly', () => {
      const tree = shallow(
        <FooterOne/>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should contain 2 input', () => {
      const tree = shallow(
        <FooterOne/>
      );
      expect(tree.find('input')).toHaveLength(2);
    })

});