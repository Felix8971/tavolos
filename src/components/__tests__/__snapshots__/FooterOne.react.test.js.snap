// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterOne should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterOne />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="item"
        >
          <h4>
            Find out about new openings and offers.
          </h4>
          <p>
            Shh...Just don't tell anyone.
          </p>
        </div>,
        <div
          className="item"
        >
          <form>
            <label>
              <input
                name="name"
                placeholder="email address"
                type="text"
              />
            </label>
            <input
              type="submit"
              value="Go"
            />
          </form>
        </div>,
      ],
      "className": "footer-one",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Find out about new openings and offers.
            </h4>,
            <p>
              Shh...Just don't tell anyone.
            </p>,
          ],
          "className": "item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Find out about new openings and offers.",
            },
            "ref": null,
            "rendered": "Find out about new openings and offers.",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Shh...Just don't tell anyone.",
            },
            "ref": null,
            "rendered": "Shh...Just don't tell anyone.",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form>
            <label>
              <input
                name="name"
                placeholder="email address"
                type="text"
              />
            </label>
            <input
              type="submit"
              value="Go"
            />
          </form>,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                <input
                  name="name"
                  placeholder="email address"
                  type="text"
                />
              </label>,
              <input
                type="submit"
                value="Go"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  name="name"
                  placeholder="email address"
                  type="text"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "name",
                  "placeholder": "email address",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Go",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="item"
          >
            <h4>
              Find out about new openings and offers.
            </h4>
            <p>
              Shh...Just don't tell anyone.
            </p>
          </div>,
          <div
            className="item"
          >
            <form>
              <label>
                <input
                  name="name"
                  placeholder="email address"
                  type="text"
                />
              </label>
              <input
                type="submit"
                value="Go"
              />
            </form>
          </div>,
        ],
        "className": "footer-one",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Find out about new openings and offers.
              </h4>,
              <p>
                Shh...Just don't tell anyone.
              </p>,
            ],
            "className": "item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Find out about new openings and offers.",
              },
              "ref": null,
              "rendered": "Find out about new openings and offers.",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Shh...Just don't tell anyone.",
              },
              "ref": null,
              "rendered": "Shh...Just don't tell anyone.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form>
              <label>
                <input
                  name="name"
                  placeholder="email address"
                  type="text"
                />
              </label>
              <input
                type="submit"
                value="Go"
              />
            </form>,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  <input
                    name="name"
                    placeholder="email address"
                    type="text"
                  />
                </label>,
                <input
                  type="submit"
                  value="Go"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    name="name"
                    placeholder="email address"
                    type="text"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "name",
                    "placeholder": "email address",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "submit",
                  "value": "Go",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
